### ISSUE TITLE
Separate external services from internal resources in the Adonis API

### ISSUE THREAD

**Issue by jonstorer** [LegalShield/styleguide#9]
we've discussed separating our API into it's parts. So far we have:
```
/user
/partner
/admin
```
I'm suggesting we add `/service` for external services we nest in our API. I suggest we nest external services, like Address Validation with a service like Smarty Streets.

Nesting the services in our API allows use to normalize different services in different countries. For example, the US may use smarty streets & the UK may use a UK provider. Normalizing the service means our clients (mobile, web) can learn a little less.

Thoughts?

**Comment by jcholder-legal** [comment]
Sounds good to me. I recommend also namespacing each service (e.g. `/services/address_validation/validate_address`) in anticipation of support routes that all services will have in common (e.g. `/services/address_validation/status`).

**Comment by jonstorer** [comment]
We're about to hit this for the UK, does anyone have an issue with doing this? @maulin @jcholder-legal @douglasrlee @danlong005

**Comment by douglasrlee** [comment]
I don't. Makes sense to me.

**Comment by maulin** [comment]
I get why you would wanna namespace /user, /partner etc, but not so sure why you'd want to make that distinction for 3rd party services? As a consumer of the api, you'd hit an endpoint that does address validation, `/address_validation`. Why does it matter if its scoped under `/services`? As a consumer I don't really care if its being serviced by Adonis or another 3rd party

**Comment by dannyburrows** [comment]
@maulin I agree that the interface should be common and each service should map to the same interface. Namespacing the implementations just further separates them and makes code / file navigation easier, in my opinion. It also helps drastically if you have to implement multiple versions of the same API (not likely but I've had to do it before).
